core/roboconf-target-iaas-ec2/src/main/java/net/roboconf/target/ec2/internal/Ec2MachineConfigurator.java

```diff
@@ -148,23 +189,26 @@ private boolean checkVmIsKnown() {
 				&& disresult.getReservations().get( 0 ).getInstances().size() > 0;
 	}
 
-
 	/**
-	 * Tags the VM (basically, it gives it a name).
+	 * Tags the specified resource, eg. a VM or volume (basically, it gives it a name).
+	 * @param resourceId The ID of the resource to tag
+	 * @param tagName The resource's name
 	 * @return true if the tag was done, false otherwise
 	 */
-	private boolean tagVm() {
-
-		// We cannot tag directly after the VM creation. See #197.
-		// We need it to be known by all the EC2 components.
-		Tag tag = new Tag( "Name", this.tagName );
-		CreateTagsRequest ctr = new CreateTagsRequest(Collections.singletonList(this.machineId), Arrays.asList( tag ));
-		this.ec2Api.createTags( ctr );
-
-		return true;
+	private boolean tagResource(String resourceId, String tagName) {
+		if(! Utils.isEmptyOrWhitespaces(tagName)) {
```

Comment:
For debug purpose, it is better when there is only one return statement in such a function.
Something like...

```java
boolean result = false;
if( ... ) {
        result = true;
}

return result;
```

In debug mode, setting a break point at the function end goes quicker than following all the possible paths.